frontend/src/components/App/DataServer/DataServerForm.tsx

```diff
@@ -0,0 +1,260 @@
+              className="form-control"
+              placeholder="e.g. '127.0.0.1'"
+              disabled={!canEdit}
+              required />
+            <Field
+              name="cephPort"
+              label="Ceph Port"
+              component={FormikInput}
+              className="form-control"
+              placeholder="e.g. '80'"
+              disabled={!canEdit}
+              required />
+            <Field
+              name="cephIsSecure"
+              label="Does Ceph use SSL?"
+              component={FormikInput}
+              type="select"
+              className="form-control"
+              placeholder="If ceph is secure, then choose 'Yes'."
+              options={yesno}
+              onChange={()=>{}}
+              disabled={!canEdit}
+              required />
+            <Field
+              name="cephBucketName"
+              label="Ceph Bucket Name"
+              component={FormikInput}
+              className="form-control"
+              placeholder="e.g. 'xxxxx'"
+              disabled={!canEdit}
+              required />
+          </CardBody>
+        </Card>
+      )
+    } else if (this.state.dataServerMode === dataServerMode.aws_s3.toString()) {
+      fields = (
+        <Card className='mb-3'>
+          <CardBody>
+            <Field
+              name="awsAccessKey"
+              label="AWS Access Key"
+              component={FormikInput}
+              className="form-control"
+              placeholder="e.g. 'xxxxx'"
+              disabled={!canEdit}
+              required />
+            <Field
+              name="awsSecretKey"
+              label="AWS Secret Key"
+              component={FormikInput}
+              className="form-control"
+              placeholder="e.g. 'xxxxx'"
+              disabled={!canEdit}
+              required />
+            <Field
+              name="awsBucketName"
+              label="AWS Bucket Name"
+              component={FormikInput}
+              className="form-control"
+              placeholder="e.g. 'xxxxx'"
+              disabled={!canEdit}
+              required />
+          </CardBody>
+        </Card>
+      )
+    } else {
+      fields = null
+    }
+
+    return (
+      <div className='pt-3 pr-3 pl-3'>
+        <h1>
+          <i className='fas fa-plug fa-fw mr-2'></i>
+          {method === 'post' ? 'Add' : 'Edit'} DataServer
+        </h1>
+        <Formik
+          initialValues={initialValues}
+          validationSchema={DataServerSchema}
+          onSubmit={onSubmit}
+          onReset={onCancel}>
+          {({ errors, touched, isSubmitting }) => (
+            <Form>
+              <Card className='mb-3'>
+                <CardBody>
+                  {this.renderModes()}
+                </CardBody>
+              </Card>
+              {fields}
+              {canEdit ?
+                <div className='text-right mb-3'>
+                  <Button color='success' type='submit' disabled={isSubmitting} >
```

Comment:
Yes. It's bothering me.
I will use `setSubmitting`! Thank you for the information!

Reply:
Done! https://github.com/rekcurd/dashboard/pull/83/commits/df5f04853e3807283b508c8b498a2e7f215c6d1e

Reply:
In that way, the button gets enabled even when the response of submission is not received.
So we should call the function in [here](https://github.com/rekcurd/dashboard/pull/83/files#diff-e2edc491134c7c0a76829521e6632f20R79), but this can be troublesome...

Reply:
I understand what you intent.
However, I am not sure how to realize it that we refer `setSubmitting` on `getDerivedStateFromProps` method.

Reply:
Hmm...
what about FormComponent also has `saveDataServerStatus` and set `disabled={isAPIFetching<boolean>(saveDataServerStatus)}` ?

Reply:
In this way, only while waiting for the response of submitting, the button will be disabled.

Reply:
I have updated -> https://github.com/rekcurd/dashboard/pull/83/commits/ed8581543ce11cbcb18f00d36121837d79407ac5

I set `setSubmitting(false)` after finishing the request by using `Promise` response.

Reply:
Nice work!! I confirmed it works